src/custom/state/swap/extension.ts

```diff
@@ -0,0 +1,159 @@
+import { CurrencyAmount, Trade, Currency } from '@uniswap/sdk'
+import { useTradeExactIn, useTradeExactOut } from 'hooks/Trades'
+import { tryParseAmount } from './hooks'
+import { basisPointsToPercent } from 'utils'
+import { FeeInformation } from 'custom/utils/operator'
+
+interface DetermineFee {
+  inputAmount: CurrencyAmount
+  feeInformation: FeeInformation
+}
+
+/**
+ * DetermineFee
+ * @description using the inputAmount, compares whether the minimalFee or the feeRatio is greater and returns
+ * @description accepts fractional values
+ */
+function determineFee({ feeInformation, inputAmount }: DetermineFee): CurrencyAmount | undefined {
+  const { feeRatio, minimalFee } = feeInformation
+
+  // We need to consider some cases here:
```

Comment:
Is it possible you created too many cases?

In my head this is only delegating to our util, and returning a new `CurrencyAmount` using that information.

It would never return undefined, we could always return a fee, even if it's 0. It feels simpler.

Reply:
removed in favour of yours!